### PR TITLE
Introduces `share` command and `list-folders` subcommand.

### PR DESCRIPTION
**Pull request by bonafidehan**

### CODE REVIEW

File: cmd/share-list-folders.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	dbx := sharing.New(config)
+	res, err := dbx.ListFolders(arg)
+	if err != nil {
+		return
+	}
+
+	// TODO(bonafidehan): handle paging. Currently uses default limit of 1000.
+
+	for _, f := range res.Entries {
+		fmt.Printf("%v\n", f.PathLower)
```
Comment (diwakergupta) [inline cmd/share-list-folders.go]: Anything more interesting we can print here? OK to do in a later commit.

File: cmd/share-list-folders.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/dropbox/sharing"
+	"github.com/spf13/cobra"
+)
+
+func shareListFolders(cmd *cobra.Command, args []string) (err error) {
+	arg := sharing.NewListFoldersArgs()
+
+	dbx := sharing.New(config)
+	res, err := dbx.ListFolders(arg)
```
Comment (diwakergupta) [inline cmd/share-list-folders.go]: Can you add some error handling here? E.g. is arg OK to be empty?

File: cmd/share-list-folders.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	for _, f := range res.Entries {
+		fmt.Printf("%v\n", f.PathLower)
+	}
+
+	return
+}
+
+var shareListCmd = &cobra.Command{
+	Use:   "list-folders",
```
Comment (diwakergupta) [inline cmd/share-list-folders.go]: `list-folders <path>`?

File: vendor/vendor.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
 	"comment": "",
-	"ignore": "appengine test",
```
Comment (diwakergupta) [inline vendor/vendor.json]: It's important to leave appengine in here. Was this an intentional change?

Comment (bonafidehan) [inline vendor/vendor.json]: This was a govender change. Will remove this file from commit.

File: cmd/share-list-folders.go (hunk 1 above)
Comment (bonafidehan) [inline cmd/share-list-folders.go]: The URL is really useful. But worried about overflowing, wish we had short(er) links.

The rest should be in `--long`.

File: cmd/share-list-folders.go (hunk 3 above)
Comment (bonafidehan) [inline cmd/share-list-folders.go]: Will add next.

File: cmd/share-list-folders.go (hunk 1 above)
Comment (bonafidehan) [inline cmd/share-list-folders.go]: Added URL. Not very pretty, especially with simple tab-delimiter, but useful > pretty.

File: cmd/share-list-folders.go (hunk 2 above)
Comment (bonafidehan) [inline cmd/share-list-folders.go]: Can you clarify what you mean by error handling? In this case, arg is `sharing.NewListFoldersArgs()`.

Comment (diwakergupta) [inline cmd/share-list-folders.go]: Yes, I meant how are we using `args` and I think you meant in the comment below that you actually plan to use it in the next PR?

**Pull request merged**